src/vs/editor/browser/widget/diffEditorWidget2/diffEditorWidget2.ts

```diff
@@ -512,6 +512,22 @@ export class DiffEditorWidget2 extends DelegatingEditor implements IDiffEditor {
 		}
 		destination.focus();
 	}
+
+	collapseAllUnchangedRegions(): void {
+		const unchangedRegions = this._diffModel.get()?.unchangedRegions.get();
+		if (!unchangedRegions) { return; }
+		for (const region of unchangedRegions) {
+			region.collapseAll(undefined);
```

Comment:
They should run in a single transaction (`transaction(tx => { /* use tx */ })`), otherwise the editor will re-layout one time for every region.

Reply:
Done